WalletWasabi/Wallets/Wallet.cs

```diff
@@ -137,7 +138,18 @@ public IEnumerable<SmartTransaction> GetTransactions()
 		}
 		return walletTransactions.OrderByBlockchain().ToList();
 	}
-	
+
+	public HdPubKey CreateReceiveAddress(IEnumerable<string> destinationLabels)
+	{
+		if (KeyManager.MasterFingerprint == null)
+		{
+			throw new InvalidOperationException("Master fingerprint should not be null");
+		}
```

Comment:
Why is generating a receive key depending on the `MasterFingerprint`?

Reply:
`MasterFingerprint` shouldn't be nullable but it is for historical reasons only. This is how it should be interpreted:

| Has EncryptedSecret? | Has MasterFingerprint | Meaning |
|--------------------------------|---------------------------------|-------------|
| YES:| YES | Normal wallet |
| NO:| YES | Hardware wallet |
| NO:| NO | ReadOnly wallet (!?)|
|YES| NO| Some idiot edited the wallet and remove it |

Now, a `readonly wallet` is something that doesn't really exist but it is something that we have to deal with because many years ago we told people that was possible to create a read only wallet just by creating a wallet json file like the following:

```
{
   "ExtPubKey" : "xpubwhatever"
}
```

Now we have to live with it or break compatibility with a feature that nobody uses and that forces me to explain the same again and again.

---------------

> Why is generating a receive key depending on the MasterFingerprint?

it makes no sense.

Reply:
Do it in WalletModel for now, it will help us to understand the rest of the code easier. We can push it down to "backend" code later.

Reply:
@soosr removed extraneous `MasterFingerprint` validation from that method. I don't recall where that was or how it ended up here.